subiquity/core.py

```diff
@@ -151,6 +151,14 @@ def make_screen(self):
             return s
 
     def run(self):
+        if not os.path.exists(self.opts.autoinstall):
+            from cloudinit import atomic_helper, stages
+            cfg = stages.Init().cfg
```

Comment:
cfg = stages.Init().read_cfg()

cfg has some redactions, not clear to me if that will impact where you're looking.

Note the Init cfg property is /etc/cloud , /etc/cloud.cfg.d/* and datasource cfg, but it is NOT user-data.

I think what you want is something like this:
```
from cloudinit import stages
init = stages.Init()
init.read_cfg()
init.fetch(existing="trust")
cloud = init.cloudify()
full_config = cloud.cfg
```
This is the config that's passed to config modules and all of the system config, user-data, vendor-data have all be processed and merged (where processed means downloading from a URL, for example).  As an example, if the user-data were a multi-part mime, then you cannot read /var/lib/cloud/instance/user-data.txt directly (without doing more work)

```
>>> from cloudinit import stages
>>> init = stages.Init()
>>> init.read_cfg()
>>> init.fetch(existing='trust')
>>> cloud = init.cloudify()
>>> cloud.datasource.userdata
<email.mime.multipart.MIMEMultipart object at 0x7f1e4dc88760>
>>> cloud.cfg
{'final_message': 'The system is finally up, after $UPTIME seconds', 'package_update': True, 'package_upgrade': True, 'packages': ['ifupdown'], 'datasource_list': ['NoCloud', 'None'],
```

 

Now, if you from know where the autoinstall cfg is being supplied (for example the kernel cmdline is writing it to a file on /etc/cloud/cloud.cfg.d/ ... then we can make some short cuts here.

Reply:
> This is the config that's passed to config modules and all of the system config, user-data, vendor-data have all be processed and merged (where processed means downloading from a URL, for example).

great, that's exactly what I want

> As an example, if the user-data were a multi-part mime, then you cannot read /var/lib/cloud/instance/user-data.txt directly (without doing more work)

yeah, this is the sort of thing I was worried about.

> Now, if you from know where the autoinstall cfg is being supplied (for example the kernel cmdline is writing it to a file on /etc/cloud/cloud.cfg.d/ ... then we can make some short cuts here.

I would prefer not to make any assumptions here if I can help it.